File: java/believe/command/CommandGenerator.java
package believe.command;

import believe.core.PropertyProvider;

import java.util.Optional;

/** Generates a {@link Command} instances when prompted. */
public interface CommandGenerator {
  /**
   * Generates a command based on {@code commandName} and the properties available in {@code
   * propertyProvider}.
   */
  Optional<Command> generateCommand(String commandName, PropertyProvider propertyProvider);
}

File: java/believe/command/CommandGenerator.kt
package believe.command

import believe.command.proto.CommandProto

/** Generates commands based on the contents of a [CommandProto.Command] proto. */
interface CommandGenerator {
    /** Returns a [Command] instance based on the contents of [commandData]. */
    fun generateCommand(commandData: CommandProto.Command): Command?
}